src/pages/event.tsx

```diff
@@ -0,0 +1,157 @@
+import { useLocale } from "@/locales";
+import { useState, useRef } from "react";
+function SingleEvent({
+  bgColor,
+  borderColor,
+  toReverse,
+  textColor,
+  image,
+  eventInfo,
+  price,
+}: {
+  bgColor: string;
+  borderColor: string;
+  toReverse: boolean;
+  textColor: string;
+  image: string;
+  eventInfo: string[];
+  price: string;
+}) {
+  const t = useLocale();
+  const [modalState, setModal] = useState(false);
+  const modalRef = useRef();
```

Comment:
The typescript compiler doesn't understand what this should be, so make this be `useRef<HTMLDivElement>(null)`.

Reply:
Fixed